bigquery/google/cloud/bigquery/client.py

```diff
@@ -133,11 +134,28 @@ def __init__(
         self._location = location
         self._default_query_job_config = default_query_job_config
 
+        # Set the USER_AGENT with the property handler for application_name.
+        self.application_name = ''
```

Comment:
This part needs assertions in one or more of the constructor unit tests (to prevent regressions).